lib/src/realm_class.dart

```diff
@@ -194,6 +194,14 @@ class Realm {
     final handle = realmCore.queryClass(this, metadata.class_.key, query, args);
     return RealmResultsInternal.create<T>(handle, this);
   }
+
+  @override
+  // ignore: hash_and_equals
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
+    if (other is! Realm) return false;
+    return realmCore.realmEquals(this, other);
```

Comment:
I think this will always return false. I cannot invent a way, where this would return true, but `identical(this, other)` would not?

Reply:
Even after exposing `RealmObject.realm`

Reply:
Are you caching realm instances at the dart layer? In most SDKs, you can get different managed instances that point to the same native one, in which case `identical` (which if I understand correctly is equivalent to `ReferenceEquals`) would return `false`.

Reply:
Correct. All calls start from a realm, and all objects created will be given this realm, even query results. Anyway, as I have implemented it here, I don't depend on it.

Reply:
I meant, like:

```dart
var config = Configuration()
  ..path = "/tmp/my-realm"

var realm1 = Realm(config);
var realm2 = Realm(config);

// I would expect
assert(identical(realm1, realm2), false);
assert(realm1 == realm2, true);
```

Reply:
As you may notice in the test that is not the case. Good to know, it is the expected behaviour.

Reply:
Yeah, this sounds like a bug in Core/the C API in that case. Those should be the same native instance due to OS caching.

Reply:
We are using realm-core v11.9.0.